src/main/webapp/app/editor/editor.service.ts

```diff
@@ -0,0 +1,51 @@
+import { Injectable } from '@angular/core';
+import { SERVER_API_URL } from '../app.constants';
+import { HttpClient, HttpParameterCodec, HttpParams } from '@angular/common/http';
+import { Cacheable } from 'ngx-cacheable';
+
+@Injectable()
+export class EditorService {
+
+    private resourceUrl =  SERVER_API_URL + 'api/plantuml';
+    private encoder: HttpParameterCodec;
+
+    constructor(private http: HttpClient) {
+        this.encoder = new HttpUrlCustomEncoder();
+    }
+
+    /**
+     * @function getPlantUmlImage
+     * @param plantUml definition obtained by parsing the README markdown file
+     * @desc Requests the plantuml png file as arraybuffer and converts it to base64
+     */
+    @Cacheable({
+        /**
+         * Cacheable configuration
+         */
+        maxCacheCount: 3,
```

Comment:
please use constants

Reply:
Need to pass the values to the method decorator directly since it has no reference to the 'this' scope and thereforce can't access local constants.